lib/src/Pos/Diffusion/Full/Block.hs

```diff
@@ -306,32 +316,42 @@ streamBlocks logTrace smM logic batchSize streamWindow enqueue nodeId tipHeader
         , mssWindow = streamWindow
         }
 
-    -- Enqueue a conversation which will attempt to stream.
-    -- This returns when the conversation is dequeued, or throws an exception
-    -- in case it's aborted or is not enqueued.
+    -- Enqueue the request blocks conversation and then
+    -- - if exception, abort it or kill it
+    -- - if no exception, return it
+    -- the `bracket` on the RHS of `streamBlocks` will kill it if this
+    -- thread is killed, and it also awaits the sub-thread so will be
+    -- killed if the sub thread dies.
```

Comment:
This looks good to me. If there is an exception during `enqueue` the `abortOrCancel` will kill the streaming thread, and if it will raise an exception it will be re-thrown by `Async.wait` in the `bracket` above.  Do we need the `Async.cancel` in the above bracket, in case when the streaming thrown an exception the thread will be dead, and if `Async.wait` finished there will also be no point in cancelling it.

Reply:
Good question. I think we do need the bracket. Is there a possibility that an exception arises in this thread after
`restore waitForDequeue `onException` abortOrCancel` has finished but before the `Async.wait` has started?

Reply:
possibly. Equivalently, we could  do `restore (waitForDequeu >>= Async.await) `onException` abortOrCancel` and then there's no need for the `bracket`.